tests/unit/test_normalize_icon_type.py

```diff
@@ -0,0 +1,65 @@
+#-----------------------------------------------------------------------------
+# Copyright (c) 2005-2022, PyInstaller Development Team.
+#
+# Distributed under the terms of the GNU General Public License (version 2
+# or later) with exception for distributing the bootloader.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+#
+# SPDX-License-Identifier: (GPL-2.0-or-later WITH Bootloader-exception)
+#-----------------------------------------------------------------------------
+
+import os
+import sys
+
+import pytest
+
+
+def test_normalize_icon_type(monkeypatch):
+    from PyInstaller.building.icon import normalize_icon_type
+
+    # Currently assumes PIL/Pillow is installed on the testing machine
+
+    workpath = data_dir = "PyInstaller/bootloader/images"
```

Comment:
`data_dir` needs to be made independent of working directory and PyInstaller installation location. That's why the Alpine test is failing. The `workpath` should be a temporary directory (see [tmp_path fixture](https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html#tmp-path)) so that we a) have test isolation and b) aren't in danger of accidentally uploading those generated images to PyPI the next time we do a release. Also, this would all be a lot neater using `pathlib.Path()` instead of `os.path`.


```suggestion
    from pathlib import Path
    import PyInstaller
    data_dir = Path(PyInstaller.__file__).with_name("bootloader") / "images"
```

Reply:
I think I've got it now.

Reply:
You're rather missing the point of `pathlib`. The idea is that you leave `data_dir` as a `Path()` then later on you can use `icon = data_dir / 'icon-console.ico'` instead of `icon = os.path.join(data_dir, 'icon-console.ico')`. But that doesn't really matter.

Reply:
I am new to pathlib, but I wanted to make sure the paths passed to the normalize function are strings, as that's what they will be in use.

Maybe pathlib paths are the same as strings for most things, but they aren't totally equivalent. In my time as a pygame contributor we've we had to put some special case code in to deal with Paths.